src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -91,6 +113,33 @@ public function generateUniqueCode()
      */
     protected function isUsedCode($code)
     {
-        return null !== $this->repository->findOneBy(['code' => $code]);
+        foreach ($this->generatedCoupons as $coupon) {
```

Comment:
But we will still have to check current part of generated coupons. So there should be 2 checks. First in database and second one in pool. But this will help us to remove this nasty `generatedCoupons` on class, and move it to method.